common/kvstore/tablestore/table_store_builder.go

```diff
@@ -0,0 +1,402 @@
+package tablestore
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/common/kvstore"
+	"github.com/Layr-Labs/eigenda/common/kvstore/leveldb"
+	"github.com/Layr-Labs/eigenda/common/kvstore/mapstore"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"math"
+	"sort"
+)
+
+// The table ID reserved for the metadata table. The metadata table is used for internal bookkeeping.
+// The following data is currently stored in the metadata table:
+//   - Schema version (in case we ever need to do a schema migration)
+//   - Table deletion markers used to detect a crash during table deletion and to complete
+//     the deletion when the store is next started.
+const metadataTableID uint32 = math.MaxUint32
+
+// The table ID reserved for the namespace table. Keys in the namespace table are table IDs (uint32)
+// and values are table names (string).
+const namespaceTableID uint32 = math.MaxUint32 - 1
```

Comment:
It's helpful to call out in comment that table name and ID can be reused after a table is completely dropped from the TableStore.

Reply:
```
// The table ID reserved for the namespace table. Keys in the namespace table are table IDs (uint32)
// and values are table names (string). Although no two tables are permitted to have share a name or a table ID,
// once a table is dropped, future tables may be instantiated with the same name and the table ID may be reused.
const namespaceTableID uint32 = math.MaxUint32 - 1
```